providers/Ototoy/mod.ts

```diff
@@ -0,0 +1,460 @@
+import type { AlbumMeta, Artist, Label as RawLabel, PackagePage, Track } from './json_types.ts';
+import type {
+	ArtistCreditName,
+	Artwork,
+	EntityId,
+	HarmonyMedium,
+	HarmonyRelease,
+	HarmonyTrack,
+	Label,
+	LinkType,
+} from '@/harmonizer/types.ts';
+import { type CacheEntry, MetadataProvider, ReleaseLookup } from '@/providers/base.ts';
+import { DurationPrecision, FeatureQuality, FeatureQualityMap } from '@/providers/features.ts';
+import { parseISODateTime, PartialDate } from '@/utils/date.ts';
+import { ProviderError, ResponseError } from '@/utils/errors.ts';
+import { DOMParser, HTMLDocument } from '@b-fuze/deno-dom';
+import { parseDuration } from '../../utils/time.ts';
+
+export default class OtotoyProvider extends MetadataProvider {
+	readonly name = 'OTOTOY';
+
+	readonly supportedUrls = new URLPattern({
+		hostname: 'ototoy.jp',
+		pathname: '/_/default/p/:id',
+	});
+
+	readonly artistUrlPattern = new URLPattern({
+		hostname: this.supportedUrls.hostname,
+		pathname: '/_/default/a/:id',
+	});
```

Comment:
We can combine those patterns (untested code):
```suggestion
	readonly supportedUrls = new URLPattern({
		hostname: 'ototoy.jp',
		pathname: '/_/default/:type(a|p)/:id(\\d+)',
	});
```
In `extractEntityFromUrl` we would call the parent method and map the letter types to "artist"/"package".